tests/seeding/test_find_or_create.py

```diff
@@ -280,3 +319,35 @@ def test_find_or_create_dataset_raise_error_exist_multiple(dataset_collection):
     # test when dataset exists multiple times and raise_error flag is on
     with pytest.raises(ValueError):
         find_or_create_dataset(dataset_collection, duplicate_name, raise_error=True)
+
+def test_create_or_update_none_found(predictor_collection):
+    # test when resource doesn't exist with listed name and check if new one is created
+    assert not [r for r in list(predictor_collection.list()) if r.name == absent_name]
+    pred = SimpleMLPredictor(name=absent_name,
+                            description = '',
+                            inputs = [],
+                            outputs = [],
+                            latent_variables = [])
+    create_or_update(predictor_collection, pred)
+    assert any([r for r in list(predictor_collection.list()) if r.name == absent_name])
+
+def test_create_or_update_unique_found(predictor_collection):
+    # test when there is a single unique resource that exists with the listed name and update
+    pred = SimpleMLPredictor(name="resource 4",
+                            description = 'I am updated!',
+                            inputs = [],
+                            outputs = [],
+                            latent_variables = [])
+    create_or_update(predictor_collection, pred)
+    updated_pred = [r for r in list(predictor_collection.list()) if r.name == "resource 4"][0]
+    assert updated_pred.description == "I am updated!"
+
+def test_create_or_update_raise_error_multiple_found(predictor_collection):
+    # test when there are multiple resources that exists with the same listed name and raise error
+    pred = SimpleMLPredictor(name="resource 1",
```

Comment:
It took me a second to recognize that `resource 1` was created twice.

Reply:
added a comment for this for clarity